google/cloud/connection_options_test.cc

```diff
@@ -166,15 +166,15 @@ TEST(ConnectionOptionsTest, UserAgentProducts) {
               ElementsAre(conn_opts.user_agent_prefix()));
 }
 
-TEST(ConnectionOptionsTest, CreateChannelArgumentsDefault) {
+TEST(ConnectionOptionsTest, CreateChannelArgumentsDefault_DISABLED) {
```

Comment:
Attempted to DISABLE this unit-test, of course will remove _DISABLEd.

Reply:
Looking...

I think this is an old test, written before we introduced the `Get*ChannelArgument()` helpers: https://github.com/googleapis/google-cloud-cpp/blob/dd4758319cdd6f3cffb5c339b07277075d79724f/google/cloud/grpc_options.h#L207-L213

I'm guessing that it segfaults because we are trying to access the wrong type (e.g. string vs. int) from this weird union thingy: https://grpc.github.io/grpc/cpp/uniongrpc__arg_1_1grpc__arg__value.html

We should use `GetStringChannelArguments()` instead of calling `actual.c_channel_args()`. I think the best thing to do is to refactor the two ChannelArguments tests. I would do this in a separate PR. It would be something like this:

```
auto actual = conn_opts.CreateChannelArguments();
auto user_agent = GetStringArgument(actual, "grpc.primary_user_agent"));
ASSERT_TRUE(user_agent.has_value());
EXPECT_THAT(*user_agent, StartsWith(conn_opts.user_agent_prefix()));
```